cmd/cat-main.go

```diff
@@ -165,11 +173,23 @@ func parseCatSyntax(ctx *cli.Context) (args []string, versionID string, timeRef
 	}
 
 	timeRef = parseRewindFlag(rewind)
+	isZip := ctx.Bool("zip")
+	startO = ctx.Int64("offset")
+	tailO = ctx.Int64("tail")
+	if tailO != 0 && startO != 0 {
+		fatalIf(errInvalidArgument().Trace(), "You cannot specify both --tail and --offset")
+	}
+	if tailO < 0 || startO < 0 {
+		fatalIf(errInvalidArgument().Trace(), "You cannot specify negative --tail or --offset")
+	}
+	if isZip && (tailO != 0 || startO != 0) {
+		fatalIf(errInvalidArgument().Trace(), "You cannot combine --zip with --tail or --offset")
+	}
 	return
 }
 
 // catURL displays contents of a URL to stdout.
-func catURL(ctx context.Context, sourceURL, sourceVersion string, timeRef time.Time, encKeyDB map[string][]prefixSSEPair, isZip bool) *probe.Error {
+func catURL(ctx context.Context, sourceURL, sourceVersion string, timeRef time.Time, encKeyDB map[string][]prefixSSEPair, isZip bool, startO, tailO int64) *probe.Error {
```

Comment:
```suggestion
func catURL(ctx context.Context, sourceURL, sourceVersion string, timeRef time.Time, encKeyDB map[string][]prefixSSEPair, isZip bool, startO, tailO int64) *probe.Error {
```

Perhaps time to convert this into `opts` struct.

Reply:
Added 👍🏼  Was getting annoyed myself.